arch/sim/include/irq.h

```diff
@@ -44,6 +44,10 @@
  * Included Files
  ****************************************************************************/
 
+#ifdef CONFIG_SIM_PREEMPTIBLE
+#  include <ucontext.h>
```

Comment:
to avoid mixing nuttx api and host api, we usually don't use the host api directly.
we have a set of special files to interact with host api. (HOSTSRCS)

Reply:
I did this because I needed to keep the ```ucontext_t``` structure inside the TCB. Do you have a better alternative ?

Reply:
Two method I can come up:
1.Define XCPTCONTEXT_REGS bigger than ucontext storage
2.Define a pointer and allocate memory dynamically
But, anyway the code build inside NuttX can't include host side header file. It is't acceptable from the techincal point.

Reply:
As I'm working on my new patch (to switch from ucontext API  to sigsetjmp/siglongjmp pair) I need to keep the sigjmp_buf inside the xcpcontext. The problem is that you can't simply use a pointer and use dynamic memory allocation because the heap memory is not initialized when up_initial_state() is called for the Idle Task. We can have an array that can hold the sigjmp_buf data but the size of this is arch dependent. Do you have a better solution to this ?

Reply:
Two methods I can imagine:
1.Call malloc/free function on host side just like you call host timer API
2.Define an array of sigjmp_buf like this:
```
  sigjmp_buf g_jmpbuf[CONFIG_MAX_TASKS];
```
and access it through pid:
```
g_jmpbuf[PIDHASH(pid))]
```
BTW, you don't need save pointer inside xcpcontext.

Reply:
That makes sense, thanks !

Reply:
BTW, you can't call NuttX malloc in up_initial_state because the critical section already enter.

Reply:
I used static allocation for this